pkg/controller/podautoscaler/horizontal.go

```diff
@@ -56,10 +56,10 @@ func calculateScaleUpLimit(currentReplicas int32) int32 {
 	return int32(math.Max(scaleUpLimitFactor*float64(currentReplicas), scaleUpLimitMinimum))
 }
 
-// ConvertToVersionVia is like api.Scheme.ConvertToVersion, but it does so via an internal version first.
-// We use it since working with v2alpha1 is convinient here, but we want to use the v1 client (and
-// can't just use the internal version).  Note that it does *not* guarantee a copy is made -- this should
-// be done separately if we need to mutate the object.
+// UnsafeConvertToVersionVia is like api.Scheme.UnsafeConvertToVersion, but it does so via an internal version first.
+// We use it since working with v2alpha1 is convenient here, but we want to use the v1 client (and
+// can't just use the internal version).  Note that conversion mutates the object, so you need to deepcopy
+// *before* you call this if the input object came out of a shared cache.
 func UnsafeConvertToVersionVia(obj runtime.Object, externalVersion schema.GroupVersion) (runtime.Object, error) {
```

Comment:
isn't there a spot in the tests that calls this?  I don't remember seeing it deep copy before calling this.  Why doesn't it need to?